docs/source/user/troubleshoot.rst

```diff
@@ -223,11 +223,174 @@ When this mode is enabled, the ``vectorize`` and ``guvectorize`` decorators will
 still result in compilation of a ufunc, as there is no straightforward pure
 Python implementation of these functions.
 
+
+Debugging JIT compiled code with GDB
+====================================
+
+Setting the ``debug`` keyword argument in the ``jit`` decorator
```

Comment:
In the cuda context, `debug` enables two compiler features:

* debuginfo: DWARF info generation
* boundcheck: raise exception on out-of-bound

(see https://github.com/numba/numba/pull/2170/files#diff-8ec5e1e22375684281bc2eb2e58cddd3R53)

In the future, it would make sense to expose per-decoration option to control opt level.  With `debug=True`, we want to change the set of optimization so that the less variables are optimized-out and less jumping around the source location when the instructions are shuffled.